include/zephyr/device.h

```diff
@@ -1080,11 +1080,22 @@ device_get_dt_nodelabels(const struct device *dev)
 			(.pm_constraints = (constraints_),))				\
 		IF_ENABLED(CONFIG_PM_POLICY_DEVICE_CONSTRAINTS,				\
 			(.pm_constraints_size = (constraints_size_),))			\
-		IF_ENABLED(CONFIG_PM_DEVICE, ({ .pm_base = (pm_),})) /**/		\
+		IF_ENABLED(CONFIG_PM_DEVICE, Z_DEVICE_INIT_PM_BASE(pm_)) /**/		\
 		IF_ENABLED(CONFIG_DEVICE_DT_METADATA,					\
 			   (.dt_meta = &Z_DEVICE_DT_METADATA_NAME_GET(dev_id_),))	\
 	}
 
+/*
+ * Anonymous unions require C11. Some pre-C11 gcc versions have early support for anonymous
+ * unions but they require these braces when combined with C99 designated initializers. For
+ * more details see https://docs.zephyrproject.org/latest/develop/languages/cpp/
+ */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__) < 201100
+#  define Z_DEVICE_INIT_PM_BASE(pm_) ({ .pm_base = (pm_),})
```

Comment:
most nested define in the code base are not indented, I would keep this aligned for consistency, up to you I guess

Reply:
True, such indentation is already done in very few places but I think it's mostly because people just don't know they can. Or they are not sure and err on the overly conservative side - even though there has never been any issue ever reported related to this. I haven't seen any Zephyr style rule about this. I'm all for consistency when there are small variations to do the same thing with none clearly better than the other one. But in this case the two alternatives are not equivalent, they are: (1) add useful information, versus (2) not add useful information.

It makes little difference here but it's _very_ helpful as soon as you have more than 2-3 levels of nesting. Just imagine not indenting actual C code. So I believe it should become more popular.